rewrite-csharp/src/main/java/org/openrewrite/csharp/service/CSharpNamingService.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.csharp.service;
+
+import org.openrewrite.internal.NameCaseConvention;
+import org.openrewrite.internal.NamingService;
+
+import java.util.regex.Pattern;
+
+public class CSharpNamingService implements NamingService {
+
+    private static final Pattern STANDARD_METHOD_NAME = Pattern.compile("^[A-Z][a-zA-Z0-9]*$");
+    private static final Pattern SNAKE_CASE = Pattern.compile("^[a-zA-Z0-9]+_\\w+$");
```

Comment:
We made some changes in https://github.com/openrewrite/rewrite/pull/4725 ; might be best to similarly test here that we use C# expected naming from a few valid but odd inputs in a unit test.